tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -52,28 +52,60 @@ def update_resource_tags(tag_action, resource, tags):
             if tag_action.manager.type == 'armresource':
                 raise NotImplementedError('Cannot tag generic ARM resources.')
 
+            # build a GenericResource object from the resource string returned by the Azure API
             az_resource = GenericResource.deserialize(resource)
             api_version = tag_action.session.resource_api_version(az_resource.id)
+
+            # set the tags for the VM
             az_resource.tags = tags
 
-            client.resources.create_or_update_by_id(resource['id'], api_version, az_resource)
+            # if the vm is a marketplace image, the plan will be set.
```

Comment:
This seems pretty specific to VMs, should you add a guard so that we only do this for VMs?